crates/primitives/src/bytecode/eof.rs

```diff
@@ -215,8 +215,10 @@ mod test {
         assert_eq!(eof.data_slice(0, 4), &[0x01, 0x02, 0x03, 0x04]);
         assert_eq!(eof.data_slice(0, 5), &[0x01, 0x02, 0x03, 0x04]);
         assert_eq!(eof.data_slice(1, 2), &[0x02, 0x03]);
-        assert_eq!(eof.data_slice(10, 2), &[]);
-        assert_eq!(eof.data_slice(1, 0), &[]);
-        assert_eq!(eof.data_slice(10, 0), &[]);
+
+        const EMPTY: &[u8] = &[];
```

Comment:
I'm unsure why, but these assertions were causing compiler errors locally due to underspecified Vec types

Reply:
Rust v1.79

Reply:
I reverted the changes.

This might merely have been an artifact of me using devcontainers to develop. The devcontainer is built with Rust v1.76, after which I manually set a new `rustup` default of v1.79. `rust-analyzer` might not be picking up this change after the container has been built.

Reply:
The same error are occurring in CI: https://github.com/bluealloy/revm/actions/runs/9600615625/job/26477283846?pr=1552

Reply:
@DaniPopes I re-reverted the commit in https://github.com/bluealloy/revm/pull/1552/commits/cef93c7c5d3c054bb22d8b0623c620575ec35c5e

It seems to be a result of conflicting implementations of a trait between `core` and `serde_json` which prevents the `assert_eq!` macro from deducing the correct type.

Reply:
Interesting, looks like this https://github.com/serde-rs/json/issues/958